File: javascript/src/Character.ts
import { CharacterType } from './CharacterType';
import { Relationship } from './Relationship';

export class Character {
  id: string;
  name: string;
  characterType: CharacterType;
  private _manager: { getRelationshipsFor(character: Character): Relationship[] } | null = null;

  constructor(id: string, name: string, characterType: CharacterType) {
    this.id = id;
    this.name = name;
    this.characterType = characterType;
  }

  setManager(manager: { getRelationshipsFor(character: Character): Relationship[] }): void {
    this._manager = manager;
  }

  getRelationships(): Relationship[] {
    if (!this._manager) return [];
    return this._manager.getRelationshipsFor(this);
  }

  toString(): string {
    return `Character(${this.id}, ${this.name}, ${this.characterType})`;
  }
}

File: python/character.py
from enum import Enum
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from .relationship_manager import RelationshipManager


class CharacterType(Enum):
    PLAYER = "Player"
    NPC = "NPC"
    VILLAIN = "Villain"
    ALLY = "Ally"
    NEUTRAL = "Neutral"
    MERCHANT = "Merchant"
    MENTOR = "Mentor"
    RIVAL = "Rival"


class Character:
    def __init__(self, character_id: str, name: str, character_type: CharacterType):
        self.id = character_id
        self.name = name
        self.character_type = character_type
        self._manager: "RelationshipManager | None" = None

    def set_manager(self, manager: "RelationshipManager") -> None:
        self._manager = manager

    def get_relationships(self):
        if self._manager is None:
            return []
        return self._manager.get_relationships_for(self)

    def __repr__(self) -> str:
        return f"Character(id={self.id!r}, name={self.name!r}, type={self.character_type.value})"
